### ISSUE TITLE
Loss test  - Learning the two-moons distribution with a normalizing flow

### ISSUE THREAD

**Issue by Justinezgh** [astrodeepnet/sbi_experiments#3]
### Tests to combine the two losses :  log likelihood and score.

**Comment by Justinezgh** [comment]
Notebook : https://colab.research.google.com/drive/1t4DaL02o31OCOFifDaQS2B1f_QN5-iRq?usp=sharing

**Comment by Justinezgh** [comment]
- 1000 updates 
- loss = - log(p(x))

![image](https://user-images.githubusercontent.com/72011736/141850322-62dfd127-78ec-479e-9eb6-6b4a142fd1b2.png)

![image](https://user-images.githubusercontent.com/72011736/141852518-61cf2bb8-4c51-488e-8097-8dd9aad19f87.png)

**Comment by Justinezgh** [comment]
- 1000 updates 
- loss = norm(score_hat - score)**2

![image](https://user-images.githubusercontent.com/72011736/141853835-dcfdb6bd-deeb-4454-9616-6a53a2b3835a.png)

![image](https://user-images.githubusercontent.com/72011736/141853854-f44ab268-5bad-4684-9816-5618ff72241c.png)

**Comment by Justinezgh** [comment]
- 1000 updates 
- loss = -log(p(x)) + mu* norm(score_hat - score)**2 

![image](https://user-images.githubusercontent.com/72011736/142002344-1fcd5a22-2200-4c90-99f9-a8acd283b9a7.png)

![image](https://user-images.githubusercontent.com/72011736/142002368-e26776d4-8e6f-4482-9539-caaec232864c.png)

![image](https://user-images.githubusercontent.com/72011736/142002416-daf0f956-3411-4a0c-9b91-5d3ec0f21c01.png)

**Comment by Justinezgh** [comment]
- 10 000 updates 
- loss = -log(p(x)) + mu* norm(score_hat - score)**2

![image](https://user-images.githubusercontent.com/72011736/142007265-8c06c7e3-98f7-489c-8ab6-f71419739e10.png)

![image](https://user-images.githubusercontent.com/72011736/142007294-dc867827-ae97-49a7-aefb-a69288622bf7.png)

![image](https://user-images.githubusercontent.com/72011736/142007316-62dadc6c-f5b3-41d2-9602-43d110bd476c.png)

**Comment by Justinezgh** [comment]
- 10 000 updates 
- loss = -nu*log(p(x)) + mu* norm(score_hat - score)**2

![image](https://user-images.githubusercontent.com/72011736/142013390-c6ce9b7d-c0e4-4f9b-a108-2541e2479713.png)

![image](https://user-images.githubusercontent.com/72011736/142013412-89ef58b9-6ba7-418b-aa63-ee449e346222.png)

![image](https://user-images.githubusercontent.com/72011736/142013435-9161ae67-b1da-4223-9f06-bb0d46f01786.png)
